ktxspec.adoc

```diff
@@ -815,38 +818,47 @@ Although other orientations can be represented, it is recommended
 that tools that create KTX files use only the values listed above
 as other values may not be widely supported by other tools.
 
-=== KTXrequiredVkExtension
+[#formatMappingMetadata]
+=== Format Mapping
+
+When <<formatMapping>> does not have an entry for the value of
+`vkFormat`, which will happen for newly addded Vulkan formats, the
+KTX writer must provide any known mapping via the following key-value
+pairs.
 
-When the value of `vkFormat` is defined by the Vulkan extension
-rather than the core specification, the required extension must be
-specified in metadata with the key
+Note that the length of these keys, including the terminating `NUL`,
+is a multiple of 4 bytes so the values will be 4-byte aligned.
 
-- `KTXrequiredVkExtension`
+==== KTXglFormat
+
+For OpenGL {,ES} the mapping is specified with the key
+
+- `KTXglFormat`
+
+The value is 12 bytes representing 3 Uint32 values:
+
+[source,c]
+----
+UInt32 glInternalformat
+UInt32 glFormat
+UInt32 glType
+----
 
-The value is the name string used needed to enable the necessary
-extension typically a string using the ASCII subset of UTF-8, e.g.,
+==== KTXdxgiFormat
 
-- `VK_IMG_format_pvrtc`
+For Direct3D the mapping is specified with the key
 
-Loaders are not required to enable the extension if they determine that
-the Vulkan implementation supports use of the format without it. This
-can happen, for example, if the implementation is of a later version of
-Vulkan where the format has become core.
+- `KTX__dxgiFormat`
 
-=== KTXrequiredVkPhysicalDeviceFeature
+The value is a UInt32 (4 bytes) giving the format enum value.
 
-Support for some standard formats is optional in Vulkan. Support
-must be queried via `VkGetPhysicalDeviceFeatures2` and, if supported,
-the feature enabled at `VkDevice` creation time. If a feature must
-be enabled to use the format given by `vkFormat`, it must specified
-in metadata with the key
+==== KTXmtlPixelFormat
 
-- `KTXrequiredVkPhysicalDeviceFeature`
+For Direct3D the mapping is specified with the key
```

Comment:
Should it be Metal here?

Reply:
Yes. Fixed in a commit I just sent. Thanks for spotting it.